### PR TITLE
Improve loc of invalid argument type errors.

### PR DESCRIPTION
**Pull request by jvilk-stripe**
Improve loc of invalid argument type errors.

Use a loc that matches relevant problematic arg rather than whole call site, if available.

### Motivation

Fixes https://github.com/sorbet/sorbet/issues/2988

6 internal users complained about this issue.

### Test plan

See included automated tests.

### CODE REVIEW

File: test/testdata/infer/star_star_args.rb [hunk 1]
```diff
@@ -38,12 +38,12 @@ def main
         one_kwarg(foo: "bad", a: "bad") # error: Expected `Integer` but found `String("bad")` for argument `foo`
         one_kwarg(foo: 1, a: 1) # error: Expected `String` but found `Integer(1)` for argument `args`
         one_kwarg(foo: "bad", a: 1) # error: Expected `Integer` but found `String("bad")` for argument `foo`
-      # ^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Expected `String` but found `Integer(1)` for argument `args`
+      #           ^^^^^^^^^^^^^^^^ error: Expected `String` but found `Integer(1)` for argument `args`
```
Comment (jez) [inline test/testdata/infer/star_star_args.rb:41]: I think if you convert some of the errors in this file from `# error:` to `# ^^^ error:` it'll be more clear that all keyword args share a single loc.

Comment (jez) [reply test/testdata/infer/star_star_args.rb:41]: Happy to be proven wrong with a test though!

Comment (jvilk-stripe) [reply test/testdata/infer/star_star_args.rb:41]: This fixes kwargs; I forgot to add my new test.

Comment (jvilk-stripe) [reply test/testdata/infer/star_star_args.rb:41]: I think the issue in this specific file is `**` parameters.

**Pull request merged**